lib/lambda_ethereum_consensus/fork_choice/helpers.ex

```diff
@@ -257,14 +265,114 @@ defmodule LambdaEthereumConsensus.ForkChoice.Helpers do
     end
   end
 
-  def state_root_by_id(id) do
-    with {:ok, {block_root, optimistic, finalized}} <- block_root_by_id(id),
+  def state_root_by_state_id(id) do
+    with {:ok, {block_root, optimistic, finalized}} <- block_root_by_block_id(id),
          {:ok, block} <- BlockStore.get_block(block_root) do
       %{message: %{state_root: state_root}} = block
       {:ok, {state_root, optimistic, finalized}}
     end
   end
 
+  @spec block_by_block_id(block_id()) ::
+          {:ok, block_info()}
+          | {:error, String.t()}
+          | :not_found
+          | :empty_slot
+          | :invalid_id
+  def block_by_block_id(:head) do
```

Comment:
Instead of having several versions of block_by_block_id, why don't we just have a single one that:

1. Gets the root for that id using. `block_root_by_block_id`.
2. Uses that root to get the block using the `BlockStore`.